subsys/bluetooth/mesh/vnd/le_pair_resp.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/random/rand32.h>
+
+#include <zephyr/bluetooth/conn.h>
+#include <bluetooth/mesh/vnd/le_pair_resp.h>
+#include "../model_utils.h"
+#include "mesh/net.h"
+#include "mesh/access.h"
+
+#define LOG_LEVEL CONFIG_BT_MESH_MODEL_LOG_LEVEL
+#include "zephyr/logging/log.h"
+LOG_MODULE_REGISTER(bt_mesh_le_pair_resp);
+
+#define BT_MESH_LE_PAIR_OP_RESET BT_MESH_MODEL_OP_3(0x11, BT_MESH_VENDOR_COMPANY_ID_LE_PAIR_RESP)
+#define BT_MESH_LE_PAIR_OP_STATUS BT_MESH_MODEL_OP_3(0x12, BT_MESH_VENDOR_COMPANY_ID_LE_PAIR_RESP)
+
+#define STATUS_PASSKEY_SET 0x00
+#define STATUS_PASSKEY_NOT_SET 0x01
+
+static int handle_reset(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
+			struct net_buf_simple *buf)
+{
+	uint32_t passkey;
+	uint8_t status = STATUS_PASSKEY_SET;
+	int err;
+
+	BT_MESH_MODEL_BUF_DEFINE(rsp, BT_MESH_LE_PAIR_OP_STATUS, 7);
+
+	passkey = sys_rand32_get() % 1000000;
+	err = bt_passkey_set(passkey);
+	if (err) {
+		LOG_ERR("Unable to set passkey (err: %d)", err);
+		status = STATUS_PASSKEY_NOT_SET;
+	}
+
+	bt_mesh_model_msg_init(&rsp, BT_MESH_LE_PAIR_OP_STATUS);
+	net_buf_simple_add_u8(&rsp, status);
+	if (status == STATUS_PASSKEY_SET) {
+		net_buf_simple_add_le48(&rsp, passkey);
```

Comment:
Packing 32 bits into 48 bits to represent 6 digits random value. Meanwhile, bt API accepts only 2 bytes. Seems there is something wrong with value width.

Reply:
Yeah, this is stupid 🤦‍♂️ . I was confused with 6 digits, it should be 4 bytes. Good catch!